internal/core/items.rs

```diff
@@ -178,6 +181,23 @@ impl ItemRc {
         r
     }
 
+    /// Return the next item in the focus chain
+    pub fn next_in_focus_chain(&self, subindex: u32) -> ItemWeak {
```

Comment:
I believe this is the case where you should use `Option<ItemRc>` instead of ItemWeak.

```suggestion
    pub fn next_in_focus_chain(&self, subindex: u32) -> Option<ItemRc> {
```